src/js/stores/__tests__/CosmosPackagesStore-test.js

```diff
@@ -306,32 +307,41 @@ describe("CosmosPackagesStore", function() {
     beforeEach(function() {
       this.requestFn = RequestUtil.json;
       RequestUtil.json = function(handlers) {
-        handlers.success(Object.assign({}, packageDescribeFixture));
+        handlers.success(Object.assign({}, serviceDescribeFixture));
       };
-      this.packageDescribeFixture = Object.assign({}, packageDescribeFixture);
+      this.serviceDescribeFixture = Object.assign({}, serviceDescribeFixture);
     });
 
     afterEach(function() {
       RequestUtil.json = this.requestFn;
     });
 
-    it("should return an instance of UniversePackage", function() {
+    it("should return an instance of UniversePackage within response", function() {
       CosmosPackagesStore.fetchServiceDescription("foo");
-      var serviceDetails = CosmosPackagesStore.getServiceDetails();
+      var response = CosmosPackagesStore.getServiceDetails();
+      var serviceDetails = new UniversePackage(response.package);
```

Comment:
This is a strange test because it's always going to pass given you created the instance here. Can you provide more clarity about the various fields `userProvidedOptions`, `resolvedOptions` and `package`?  It seems to me `CosmosPackagesStore#getServiceDetails` should return an instance of `UniversePackage` where the `resolved` data has been merged. Is that something that makes sense based on what Cosmos is returning?

Reply:
We talked offline a bit this morning. We want to make sure our store/action design is compatible with framework upgrades in the future. A few conclusions/facts:
1. Editing a framework in the UI involves populating a form with the JSON schema data. This data is available in the `package` field of the response from either `/service/describe` or the existing `package/describe`. The body of these endpoints is different because `service/describe` takes an `appId` in the body, whereas `package/describe` accepts the name and version.
2. `service/describe` also includes the additional data in the response under the key `resolvedOptions`. This is the resolution of the framework defaults and the user provided options. This is the form state when editing the framework in the UI.
3. The `Framework` struct is an appropriate place to hold the `resolvedOptions` data (it represents the current running configuration of the framework).
4. Ideally we only want a single action for getting the JSON schema for a package and returning a `UniversePackage`. This should be `package/describe`, because the naming semantically makes sense and avoids conflating the terms service and package.

At this point, we are still undecided how to move forward with this.